src/main/scala/org/broadinstitute/dsde/rawls/expressions/SlickExpressionParsing.scala

```diff
@@ -0,0 +1,302 @@
+    for {
+      workspace <- workspaceQuery.findByIdQuery(context.workspaceContext.workspaceId)
+      workspaceAttribute <- workspaceAttributeQuery if workspaceAttribute.workspaceId === workspace.id
+      attribute <- attributeQuery if workspaceAttribute.attributeId === attribute.id && attribute.name === entityRefName
+      nextEntity <- entityQuery if attribute.valueEntityRef === nextEntity.id
+    } yield (nextEntity, attribute.listIndex.getOrElse(0))
+  }
+
+  // add pipe to an entity referenced by the current entity
+  private def entityNameAttributePipeFunc(entityRefName: String)(context: SlickExpressionContext, queryPipeline: PipeType): PipeType = {
+      for {
+      (entity, listIndex) <- queryPipeline
+      entityAttribute <- entityAttributeQuery if entity.id === entityAttribute.entityId
+      attribute <- attributeQuery if entityAttribute.attributeId === attribute.id && attribute.name === entityRefName
+      nextEntity <- entityQuery if attribute.valueEntityRef === nextEntity.id
+    } yield (nextEntity, listIndex * 10 + attribute.listIndex.getOrElse(0))
+  }
+
+  // filter attributes to only the given attributeName and convert to attribute
+  private def entityAttributeFinalFunc(attributeName: String)(context: SlickExpressionContext, queryPipeline: Option[PipeType]): ReadAction[Iterable[Attribute]] = {
+    // attributeForNameQuery will only contain attribute records of the given name but for possibly more than 1 entity
+    // and in the case of a list there will be more than one attribute record for an entity
+    val attributeForNameQuery = (for {
+      (entity, listIndex) <- queryPipeline.get
+      entityAttribute <- entityAttributeQuery if entity.id === entityAttribute.entityId
+      attribute <- attributeQuery if entityAttribute.attributeId === attribute.id && attribute.name === attributeName
+    } yield (attribute, listIndex)).result
+
+
+    attributeForNameQuery.map { entityWithAttributeRecs =>
+      // split up the attribute records by entity and strip out the junk we don't care about
+      val attributeRecsGroupedByEntity = entityWithAttributeRecs.groupBy(rec => (rec._1, rec._2)).toSeq.sortBy(_._1._2).map(_._2).map(_.map { case (attributeRec, listIndex) => (attributeRec, None) })
+
+      // unmarshall all the records, unmarshall returns a map keyed by attribute name, there should be at most 1 attribute name
+      // because attributeForNameQuery applied that filter. There may be none in which case the attribute does not exist
+      // for the entity
+      attributeRecsGroupedByEntity.map(attributeQuery.unmarshalAttributes).map(_.getOrElse(attributeName, AttributeNull))
+    }
+  }
+
+  // final func that handles reserved attributes on an entity
+  private def entityReservedAttributeFinalFunc(attributeName: String)(context: SlickExpressionContext, queryPipeline: Option[PipeType]): ReadAction[Iterable[Attribute]] = {
```

Comment:
Since we are no longer using the `reservedAttributeNames` directly (which I am ok with), add a comment to the `reservedAttributeNames` definition explaining that these 2 functions will need to be updated if it changes.